sunpy/map/mapbase.py

```diff
@@ -783,10 +792,8 @@ def waveunit(self):
 
         This is taken from the 'WAVEUNIT' FITS keyword.
         """
-        unit = self.meta.get("waveunit")
-        if unit is None:
-            return u.one
-        return u.Unit(unit)
+        if 'waveunit' in self.meta:
```

Comment:
This will break wavelength if `'wavelnth'` is set and waveunit isn't?

Reply:
I think this default actually makes sense, `u.one` is a better "null" value than `None` for a unit?

Reply:
I did think about that, but a dimensionless unit is still a unit and not really a null value.

Reply:
We could have this return None but then all the other things that rely on it need changing to not explode if it's None.